include/mbedtls/ssl.h

```diff
@@ -3908,10 +3927,44 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
  *                 query the active maximum fragment length.
  *
  * \note           Attempting to write 0 bytes will result in an empty TLS
- *                 application record being sent.
+ *                 application record being prepared.
  */
 int mbedtls_ssl_write( mbedtls_ssl_context *ssl, const unsigned char *buf, size_t len );
 
+/**
+ * \brief          Dispatch all previously queued application data
```

Comment:
Not _all_ data, since `WANT_WRITE` is possible. It just sends what it can. To send all data, you need to call `flush` in a loop until it stops returning a “try again” error.

Reply:
That's true, but for the sake of the brief description, don't we usally omit the "Try to ..." prefix and explain the ways a function can fail in the main body of the documentation?

Reply:
For a function that either does the job or fails in a non-recoverable way, yes. But a `WANT_WRITE` return here is really a partial success: some progress has been made and normal operation can continue. So it's misleading to only refer to the complete success case.